nUpdate.UpdateInstaller/Core/ProgressReporterServiceEventLog.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using nUpdate.UpdateInstaller.Client.GuiInterface;
+
+namespace nUpdate.UpdateInstaller.Core
+{
+    public class ProgressReporterServiceEventLog : IProgressReporter
+    {
+        private bool shouldRun = true;
+
+ 
+        public void Fail(Exception ex)
+        {
+            WindowsEventLog.LogException(ex);
+        }
+
+        public void Initialize()
+        {
+            if(WindowsServiceHelper.IsRunningInServiceContext)
+            {
+                while (shouldRun)
+                {
+                    System.Threading.Thread.Sleep(2000);

```

Comment:
What is that for?

Reply:
in desktop mode you open a window that prevent the application to close and do async update

in service mode, the window doesent exists, so we need to loop until the update is done

Reply:
Ah, I see. Thought of something else here. Thanks.